gnowsys-ndf/gnowsys_ndf/ndf/views/event.py

```diff
@@ -257,22 +266,141 @@ def event_detail(request, group_id, app_id=None, app_set_id=None, app_set_instan
   else:
     Add="Stop"       
   #fecth the data
-      
-  context_variables = { 'groupid': group_id, 
-                        'app_id': app_id,'app_collection_set': app_collection_set, 
-                        'app_set_id': app_set_id,
-                        'title':title,
-                        'nodes': nodes, 'node': node,
-                        'event_gst':event_gst.name,
-                        'Add':Add,
-                        'reschedule_time' : reschedule_time,
-                        'reschedule'    : reschedule, 
-                        'task_date' : event_task_date_reschedule,
-                        'task_attendance' : event_task_Attendance_reschedule,
-                        'marks_entry_completed' :marks_entry_completed,
-                        'Eventtype':Eventtype, 
-                         # 'property_order_list': property_order_list
-                      }
+
+  if app_set_instance_id: #only if we view details of the particular event
+    if request.user:
+      usr = node_collection.one({'_type':u'Author','name':unicode(request.user)})
+      usrid = usr._id
+
+    for i in node.attribute_set:
+      try:
+        start_time = i['start_time']
+        break
+      except:
+        pass
+    # st_time = node.attribute_set[0]['start_time']
+    
+    for i in node.attribute_set:
+      try:
+        end_time = i['end_time']
+        break
+      except:
+        pass
+    
+    # end_time = node.attribute_set[1]['end_time']
+    
+    now = datetime.datetime.now()
+
+    buff_start = datetime.timedelta(hours=24)
+
+    beg = start_time - buff_start
+    end = end_time
+    days_left = 0
+
+    if now <= end and beg <= now :
+      active =  0
```

Comment:
I think I need at least three states to represent before event, during event and after event.